momepy/functional/_elements.py

```diff
@@ -343,3 +345,89 @@ def get_nearest_street(
 
     ids[blg_idx] = streets.index[str_idx]
     return ids
+
+
+def generate_blocks(
+    tessellation: GeoDataFrame,
+    edges: GeoDataFrame,
+    buildings: GeoDataFrame,
```

Comment:
We don't need this as it can be retrieved from tessellation which should have an index matching with buildings. a simple assignment would do the trick there.